### PR TITLE
bpf: Refactor BPF datapath to cleanly separate static data from preprocessor conditionals

### PR DESCRIPTION
**Pull request by joestringer**
Refactor use of preprocessor macros that define static data such as IPs, LXC ids, etc. such that they are only used within datapath code as data values, and not to conditionally enable code paths.

Related: #6618

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/6681)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

File: pkg/endpoint/bpf.go
Comment (ianvernon) [inline pkg/endpoint/bpf.go:228]: Does this new flag have to be added to the `lxc_config.h` for the verifier testing?

**Review by ianvernon — COMMENTED** [review]
LGTM from janitor / endpoint side.

**Review by ianvernon — APPROVED** [review]

**Comment by aanm** [comment]
please hold merging until #6653 is merged

**Comment by joestringer** [comment]
test-me-please

EDIT: timed out, just like a bunch of recent builds. I plan to wait for #6693 then rebase and hopefully by then the CI is in better shape.

**Comment by jrfastab** [comment]
test-me-please

**Comment by jrfastab** [comment]
^^^^ ah sorry about that wrong screen.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

EDIT: Timed out waiting for pods?
https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/9642/testReport/junit/k8s-1/13/K8sDatapathConfig_DirectRouting_Check_connectivity_with_automatic_direct_nodes_routes/

```
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:408
Pods are not ready after timeout
Expected
    <*errors.errorString | 0xc433810650>: {
        s: "Timeout reached: timed out waiting for pods with filter  to be ready",
    }
to be nil
/home/jenkins/workspace/Cilium-PR-Ginkgo-Tests-Validated/src/github.com/cilium/cilium/test/k8sT/DatapathConfiguration.go:148
```

Filed #6947.

**Comment by joestringer** [comment]
test-me-please

EDIT: Hit similar failures to #6957.

**Comment by joestringer** [comment]
test-me-please

EDIT: Green build

**Review by vadorovsky — APPROVED** [review]

**Comment by joestringer** [comment]
test-me-please

File: bpf/lib/eps.h
Comment (tgraf) [inline bpf/lib/eps.h:101]: Please explain in the commit message why removal of the `#if` is OK

**Review by tgraf — CHANGES_REQUESTED** [review]

File: bpf/lib/eps.h
Comment (joestringer) [reply bpf/lib/eps.h:101]: From the new commit message:

```
    The wrappers around lookup_ip*_remote_endpoint() wrap the definition of
    macros to rewrite lookups into the ipcache map. In the non-LPM case,
    the IPCACHE_PREFIXES macro must be defined to ensure that the lookups
    find all ipcache entries; in the LPM case, so long as the macro is being
    used it must be defined. If it is defined and not used, the compiler
    should compile out the unnecessary code.
    
    The policy_can_egress*() functions only called from the egress BPF
    program which is attached to lxc devices, so the LXC_ID check there was
    a no-op.
    
    These changes should have no functional impact.
```

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**
